### PR TITLE
Apply eslint-rule array-callback-return - Related to #1597

### LINKED ISSUE #1597: Enable pending eslint rules
### Expected behavior
Turned off `eslint` rules should be fixed to ensure code quality and to be able to use latest ECMA 6 features.
### Actual behavior
Bunch of `eslint` rules were turned off during the `eslint` integration to `1.0.0`, the mocha test specific eslint rules were fixed here  #1485. In root there are bunch of eslint rules still disabled, following are list of root eslint rules with number of errors.

## Manual fix
- [ ] "no-unused-expressions": 1252
- [ ] "camelcase": 813
- [ ] "no-param-reassign": 470
- [ ] "prefer-destructuring": 296
- [ ] "no-underscore-dangle": 291
- [ ] "no-plusplus": 154
- [ ] "max-len": 153
- [ ] "radix": 76
- [ ] "no-use-before-define": 76
- [ ] "no-new": 26

## Warnings
- [ ] "func-names": 1016

## DONE
- [x] "array-callback-return": #2563 
- [x] "callback-return": #2561 
- [x] "consistent-return": #2565 
- [x] "eqeqeq": #2562 
- ~"global-require": 69~ to keep until `app.js` is refactored
- [x] "guard-for-in": 3 #2648
- ~"generator-star-spacing": 2~ overwritten with Prettier
- [x] "new-cap": #2566
- [X] "no-bitwise": 13 #2590
- [X] "no-nested-ternary": 8 #2589
- [x] "no-prototype-builtins": 6 #2646
- [X] "no-restricted-globals": 9 #2649
- [x] "no-restricted-syntax": 7 #2643 
- [x] "no-shadow": 321 #2583
- ~"no-template-curly-in-string": 58~ to keep until https://github.com/LiskHQ/lisk/milestone/35 is accomplished
- ~"no-throw-literal": 29~ to keep
- [x] "prefer-rest-params": 5 #2640
- ~"strict": 286 #2585~
- [x] "vars-on-top": , 0 #2649
- ~"wrap-iife"~ overwritten with Prettier

## Auto fix using `npm run lint:fix`
- ~"space-before-function-paren": 2180~ https://github.com/prettier/prettier/issues/3847
- [X] "no-var": 4519 #2603

### Steps to reproduce
In root `.eslintrc.json` under `rules` section turn on the rules and run `npm run lint` to see the errors/warnings

### PR DESCRIPTION
**Pull request by diego-G**
### Review checklist

* The PR is related to #1597
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: db/repos/transactions/transfer.js [hunk 1]
```diff
@@ -60,14 +60,12 @@ class TransferTransactionsRepository {
...
 
-			transactions = transactions.map(transaction => {
-				if (transaction.asset && transaction.asset.data) {
-					return {
-						transactionId: transaction.id,
-						data: Buffer.from(transaction.asset.data, 'utf8'),
-					};
-				}
-			});
+			transactions = transactions
```
Comment (ManuGowda) [inline db/repos/transactions/transfer.js]: ❤️

File: test/functional/http/get/delegates.js [hunk 2]
```diff
@@ -313,7 +313,7 @@ describe('GET /delegates', () => {
 					.then(res => {
 						expect(res.body.data).to.have.length(13);
 						res.body.data.map(d => {
-							expect(/^genesis_1.*/.test(d.username)).to.be.true;
+							return expect(/^genesis_1.*/.test(d.username)).to.be.true;
```
Comment (ManuGowda) [inline test/functional/http/get/delegates.js]: you can do the same here `res.body.data.map(d => expect(/99/.test(d.username)).to.be.true);`

File: test/functional/http/get/delegates.js [hunk 3]
```diff
@@ -324,7 +324,7 @@ describe('GET /delegates', () => {
 					.then(res => {
 						expect(res.body.data).to.have.length(3);
 						res.body.data.map(d => {
-							expect(/^genesis_10.*/.test(d.username)).to.be.true;
+							return expect(/^genesis_10.*/.test(d.username)).to.be.true;
```
Comment (ManuGowda) [inline test/functional/http/get/delegates.js]: same as above comment and wherever applicable below.

File: test/functional/http/get/node/transactions_unsigned.js [hunk 4]
```diff
@@ -271,7 +273,9 @@ describe('GET /api/node', () => {
 						expect(res.body.data).to.not.empty;
 						expect(res.body.data.length).to.be.at.least(numOfTransactions);
 						res.body.data.map(transaction => {
-							expect(transaction.senderId).to.be.equal(senderAccount.address);
+							return expect(transaction.senderId).to.be.equal(
```
Comment (ManuGowda) [inline test/functional/http/get/node/transactions_unsigned.js]: Basically in es6 wherever we have a single statement in a function we do not need to use curly braces and return, you can just remove curly braces and return. I am not leaving the comment everywhere so please consider the same for other places too.

Comment (diego-G) [reply test/functional/http/get/node/transactions_unsigned.js]: I'll try to find another rule for this. There are too many cases.

**Pull request merged**